xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-ui/src/main/resources/XWiki/Notifications/Code/NotificationsDisplayerUIX.xml

```diff
@@ -322,7 +322,7 @@ Get the actual notifications.
     }
     var markAllReadButton = $('&lt;a href="#"&gt;')
       .addClass('notification-event-clean')
-      .html('$services.icon.renderHTML("trash")&amp;nbsp;$escapetool.xml($escapetool.javascript($services.localization.render('notifications.menu.clear')))')
+      .html('$escapetool.xml($services.icon.renderHTML("trash"))&amp;nbsp;$escapetool.xml($escapetool.javascript($services.localization.render('notifications.menu.clear')))')
```

Comment:
If we're rendering the icon as HTML it means we need that HTML. Moreover, we're injecting that rendered icon HTML using jQuery, so my understanding is that the rendered icon is supposed to be used as HTML not as plain text, so I don't see why you would escape the HTML. What you need to JavaScript escaping, to make sure the output of the Velocity code doesn't break the JavaScript code.

```suggestion
      .html($jsontool.serialize("$services.icon.renderHTML('trash')&amp;nbsp;$escapetool.xml($services.localization.render('notifications.menu.clear'))"))
```

Reply:
+1
I've tested the PR on a running instance and this is currently not fixing the issues with the notif and the tour.
Also, I don't know if this is related but the document tree can display strangely from time to time:
![image](https://github.com/xwiki/xwiki-platform/assets/327856/f9fe1c5a-41ee-47a7-87d1-16c46a1dc410)

Reply:
The tree fails to open at

> +1 I've tested the PR on a running instance and this is currently not fixing the issues with the notif and the tour. Also, I don't know if this is related but the document tree can display strangely from time to time: ![image](https://user-images.githubusercontent.com/327856/281015017-f9fe1c5a-41ee-47a7-87d1-16c46a1dc410.png)

This UI issue is not related to this regression, see https://github.com/xwiki/xwiki-platform/pull/2583 :)

Reply:
> If we're rendering the icon as HTML it means we need that HTML. Moreover, we're injecting that rendered icon HTML using jQuery, so my understanding is that the rendered icon is supposed to be used as HTML not as plain text, so I don't see why you would escape the HTML. What you need is JavaScript escaping, to make sure the output of the Velocity code doesn't break the JavaScript code.

Indeed, with proper tests, it's easy to see the solution I proposed before was not good:
![wrongEscaping__21475](https://github.com/xwiki/xwiki-platform/assets/28761965/7fe2a7f8-608b-4f2d-aa8f-dadce094c59a)

Proposed changes merged to the PR :+1: